### DISCUSSION TITLE
Create a shopify like product variants combination

### DISCUSSION THREAD

**Discussion by kaisiant** [react-hook-form/react-hook-form#7882] [Q&A]
I am trying to create a product variants combination that is similar to Shopify.

Currently, I am facing the problem of persisting the values of the row when changing/deleting the option.

1. Add option name "Color", option values ["Red", "Green"]
2. Add another option name "Weight", option values ["1kg"]
3. Set the red / 1kg "1" and "2"
4. Set the green / 1kg "3" and "4"
5. Add another option value "2kg" under "Weight"
6. Expected to see red / 2kg and green / 2kg with empty values

My current logic is to find the existing variants and reset the new variants. It is not perfect and causes lots of bugs.

Just wondering is there any other way to do it in the react hook form?

CSB: https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-forked-vtxqwp?file=/src/Variants.js

**Comment by mustafamoe** [comment]
I would love to know how to do it on react hook form too!!!
